### PR TITLE
[MM-55323] Allow end users to fetch the group members list of groups which allow @-mentions

### PR DESCRIPTION
**Pull request by hanzei**
#### Summary
If @-mentions are enabled for an LDAP group, end users should be able to fetch the member list so they know who is getting @-mentioned. This matches the wording on the system console.
Screenshot from 2024-03-22 11-16-11

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-55323

#### Screenshots
|  before  |  after  |
|----|----|
| Screenshot from 2024-03-22 10-57-59 | Screenshot from 2024-03-22 11-14-05  |

#### Release Note
```release-note
Allow end users to fetch the group members list of groups which allow @-mentions
```

### CODE REVIEW

File: server/channels/api4/group_test.go [hunk 1]
```diff
@@ -1475,6 +1475,66 @@ func TestGetGroupsByUserId(t *testing.T) {
...
+	th.App.Srv().SetLicense(model.NewTestLicense("ldap"))
+
+	t.Run("Non admins are not allowed to get members for LDAP groups", func(t *testing.T) {
+		members, response, err := th.Client.GetGroupMembers(context.Background(), group.Id)
+		assert.Error(t, err)
+		CheckForbiddenStatus(t, response)
+		assert.Nil(t, members)
+	})
+
+	t.Run("Admins are not allowed to get members for LDAP groups", func(t *testing.T) {
```
Comment (isacikgoz) [inline server/channels/api4/group_test.go:1517]:
```suggestion
	t.Run("Admins are allowed to get members for LDAP groups", func(t *testing.T) {
```

File: server/channels/api4/group_test.go [hunk 2]
```diff
@@ -1475,6 +1475,66 @@ func TestGetGroupsByUserId(t *testing.T) {
...
+		members, response, err := th.SystemAdminClient.GetGroupMembers(context.Background(), group.Id)
```
Comment (isacikgoz) [inline server/channels/api4/group_test.go:1518]: Should we also test with `LocalClient`

File: server/channels/api4/group_test.go [hunk 3]
```diff
@@ -1475,6 +1475,66 @@ func TestGetGroupsByUserId(t *testing.T) {
...
+		CheckOKStatus(t, response)
+		require.NotNil(t, members)
+		assert.Equal(t, 2, members.Count)
+	})
+
+	group.AllowReference = valTrue
+	group, appErr = th.App.UpdateGroup(group)
+	assert.Nil(t, appErr)
+
+	t.Run("Non admins are not allowed to get members for LDAP groups", func(t *testing.T) {
```
Comment (isacikgoz) [inline server/channels/api4/group_test.go:1529]:
```suggestion
	t.Run("Allow reference to get members for LDAP groups", func(t *testing.T) {
```

File: server/channels/api4/group_test.go [hunk 4]
```diff
@@ -1475,6 +1475,66 @@ func TestGetGroupsByUserId(t *testing.T) {
...
+		members, response, err := th.SystemAdminClient.GetGroupMembers(context.Background(), group.Id)
+		assert.NoError(t, err)
+		CheckOKStatus(t, response)
+		require.NotNil(t, members)
+		assert.Equal(t, 2, members.Count)
+	})
+
+	group.AllowReference = valTrue
+	group, appErr = th.App.UpdateGroup(group)
+	assert.Nil(t, appErr)
```
Comment (isacikgoz) [inline server/channels/api4/group_test.go:1527]: Maybe these should get into the t.Run function below. 0/5

**Review by isacikgoz — CHANGES_REQUESTED** [review]
Looks good! Just a few naming change suggestions and ready to go from my side.

File: server/channels/api4/group_test.go (hunk 2 above)
Comment (hanzei) [reply server/channels/api4/group_test.go:1518]: No, as the `/members` endpoint is not implemented for the local API.

**Review by isacikgoz — APPROVED** [review]

**Review by esarafianou — APPROVED** [review]

**Comment by hanzei** [comment]
@coltoneshaw Will cherry-pick the PR once it's merged.

**Review by BenCookie95 — APPROVED** [review]

**Comment by hanzei** [comment]
@DHaussermann Ready for your review. QA tests are in the JIRA ticket.

**Review by DHaussermann — APPROVED** [review]
Tested and passed
- Followed the steps in Jira ticket and confirmed non-sysadmin users can see the users in the pop-over for LDAP groups
- As a precaution - tested generic groups as well with no association to LDAP to see if any issues would surface.
No issues found. 

LGTM! 
@hanzei please merge.

**Pull request merged**